CMake/FindArmadillo.cmake

```diff
@@ -91,12 +91,13 @@ if(NOT _ARMA_USE_WRAPPER OR MSVC)
     endif()
   endif()
   if(_ARMA_USE_BLAS)
-    if(ARMADILLO_FIND_QUIETLY OR NOT ARMADILLO_FIND_REQUIRED)
-      find_package(BLAS QUIET)
+    if(NOT BLAS_FOUND)
+      if(ARMADILLO_FIND_QUIETLY OR NOT ARMADILLO_FIND_REQUIRED)
+        find_package(BLAS QUIET)
+      else()
+        find_package(BLAS REQUIRED)
+      endif()
```

Comment:
@rcurtin, I have found the issue that stopping it from running on CI. if we compiled `OpenBLAS` manually, we need to `set(BLAS_FOUND ON)`.
The issue is related to `find_package(BLAS)` here, since it is going to search for BLAS inside the `sysroot` and will never find it (since it is in deps), then it will set `(BLAS_FOUND OFF)` stopping it from being added to `ARMADILLO_LIBRARIES`
The solution I found is just to check whether `BLAS_FOUND` is already `ON`. In this case, there is no need to run `find_package(BLAS)`

Reply:
This doesn't seem right.  What is the version of CMake on CI?  Do you need to set `BLAS_LIBRARIES` instead of `BLAS_openblas_LIBRARIES`?  It shouldn't be necessary to make these changes, since they intertwine functionality of the cross-compilation CMake configuration and this.

Reply:
The version of CMake on CI is 3.13.4

This is the only solution I have found, but I am open to find a better solution in order to reduce the modifications to cmake files :+1:.

After hours of digging, I have discovered that it is not related to the version of cmake, even I was able to reproduce the same issue on my machine. It will work only using `BLAS_LIBRARIES` when I run cmake twice (first time failing BLAS not found) and keep the cache and then run `find_package(BLAS)`.

I do think you will face the same issue if you remove OpenBLAS from the sysroot, let me know if this is not the case.

If my understanding is right `find_package(BLAS)` will start by resetting the value of `BLAS_LIBRARIES` https://gitlab.kitware.com/cmake/cmake/blob/master/Modules/FindBLAS.cmake#L283

Then it will look inside the sysroot (when cross comiling), if it is not there it will not find it.

I have already set `BLAS_LIBRARIES` and `BLAS_openblas_LIBRARIES` which gave the same result.

I happy to remove these changes and run it on CI a second time if you like in order to repoduce the issue.

Let me know what do you think :+1: